impeller/entity/contents/filters/directional_gaussian_blur_filter_contents_unittests.cc

```diff
@@ -0,0 +1,185 @@
+          auto result = std::make_shared<MockCommandBuffer>(mock_context);
+          EXPECT_CALL(*result, IsValid()).WillRepeatedly(Return(true));
+          EXPECT_CALL(*result, OnSubmitCommands(_))
+              .WillRepeatedly(Return(true));
+          EXPECT_CALL(*result, OnCreateRenderPass(_))
+              .WillRepeatedly(
+                  Invoke(([mock_context](const RenderTarget& render_target) {
+                    auto result = std::make_shared<MockRenderPass>(
+                        mock_context, render_target);
+                    EXPECT_CALL(*result, IsValid).WillRepeatedly(Return(true));
+                    EXPECT_CALL(*result, OnEncodeCommands(_))
+                        .WillRepeatedly(Return(true));
+                    return result;
+                  })));
+          return result;
+        })));
+    EXPECT_CALL(*mock_render_target_allocator_, CreateTexture(_))
+        .WillRepeatedly(Invoke(([](const TextureDescriptor& desc) {
+          auto result = std::make_shared<MockTexture>(desc);
+          EXPECT_CALL(*result, IsValid()).WillRepeatedly(Return(true));
+          return result;
+        })));
+  }
+
+  std::shared_ptr<MockImpellerContext> mock_context_ =
+      std::make_shared<MockImpellerContext>();
+  std::shared_ptr<MockCapabilities> mock_capabilities_ =
+      std::make_shared<MockCapabilities>();
+  std::shared_ptr<MockSamplerLibrary> mock_sampler_library_ =
+      std::make_shared<MockSamplerLibrary>();
+  std::shared_ptr<MockShaderLibrary> mock_shader_library_ =
+      std::make_shared<MockShaderLibrary>();
+  std::shared_ptr<MockPipelineLibrary> mock_pipeline_library_ =
+      std::make_shared<MockPipelineLibrary>();
+  std::shared_ptr<MockTypographerContext> mock_typographer_context_ =
+      std::make_shared<MockTypographerContext>();
+  std::shared_ptr<MockAllocator> mock_allocator_ =
+      std::make_shared<MockAllocator>();
+  std::shared_ptr<MockRenderTargetAllocator> mock_render_target_allocator_;
+  std::shared_ptr<const Capabilities> capabilities_;
+};
+
+TEST_F(DirectionalGaussianBlurFilterContentsTest, CoverageWithEffectTransform) {
+  TextureDescriptor desc = {
+      .size = ISize(100, 100),
+  };
+  Scalar sigma_radius_1 = CalculateSigmaForBlurRadius(1.0);
+  auto contents = std::make_unique<DirectionalGaussianBlurFilterContents>();
+  contents->SetSigma(Sigma{sigma_radius_1});
+  contents->SetDirection({1.0, 0.0});
+  std::shared_ptr<MockTexture> texture = std::make_shared<MockTexture>(desc);
+  EXPECT_CALL(*texture, GetSize()).WillRepeatedly(Return(ISize(100, 100)));
+  FilterInput::Vector inputs = {FilterInput::Make(texture)};
+  Entity entity;
+  entity.SetTransformation(Matrix::MakeTranslation({100, 100, 0}));
+  std::optional<Rect> coverage = contents->GetFilterCoverage(
+      inputs, entity, /*effect_transform=*/Matrix::MakeScale({2.0, 2.0, 1.0}));
+  ASSERT_EQ(coverage, Rect::MakeLTRB(100 - 2, 100, 200 + 2, 200));
```

Comment:
Here and elsewhere:

I might either expand on the test name, or leave a comment (or does ASSERT support reasons, i.e. << "Reason") of why this is the expected result. From someone totally unfamiliar with this code, I don't understand what the test would mean if it failed or why this is the expected result.